### ISSUE TITLE
InternalExpectAsync does not await actionAsync() - causing actionAsync to run as a detached task

### ISSUE THREAD

**Issue by brah-mcdude** [akkadotnet/akka.net#5537]
why does this pass:

```csharp
/// <summary>
/// this test is a false positive
/// </summary>
/// <returns></returns>
[Fact]
public async Task ReplyingToProbeMessagesFunc()
{
    await EventFilter.Error().ExpectAsync(0, actionAsync: async () =>
    {
        var probe = CreateTestProbe();
        probe.Tell("hello");
        probe.ExpectMsg("hello");
        probe.Reply("world");
        await Task.Run(() => { ExpectMsg("world2"); });
        Assert.Equal(probe.Ref, LastSender);
    });
}
```

Looks to me like we forgot to.... you know, `await` the `Task` returned inside here: 
https://github.com/akkadotnet/akka.net/blob/38b7443fd03f0025ac0e7a7a1ed5e1f636bfb6ba/src/core/Akka.TestKit/EventFilter/Internal/EventFilterApplier.cs#L454-L460

So I suspect the ` await Task.Run(() => { ExpectMsg("world2"); });` is running as a detatched task, which is incredibly annoying. We just need to update the TestKit to await those tasks internally and then this should pass.

_Originally posted by @Aaronontheweb in https://github.com/akkadotnet/akka.net/issues/5529#issuecomment-1020502375_

**Comment by Aaronontheweb** [comment]
Just need to `await actionAsync()` and delete the `Task.FromResult` call there I think.

**Comment by brah-mcdude** [comment]
```c#
Akka.TestKit.Xunit2.Internals.AkkaEqualException
  HResult=0x80131500
  Message=Assert.Equal() Failure
Expected: world2
Actual:   world
  Source=Akka.TestKit.Xunit2
  StackTrace:
   at Akka.TestKit.Xunit2.XunitAssertions.AssertEqual[T](T expected, T actual, String format, Object[] args) in C:\Users\brahm\source\repos\akka.net\src\contrib\testkits\Akka.TestKit.Xunit2\XunitAssertions.cs:line 68
   at Akka.TestKit.TestKitBase.<>c__DisplayClass108_0`1.<ExpectMsg>b__0(T m) in C:\Users\brahm\source\repos\akka.net\src\core\Akka.TestKit\TestKitBase_Expect.cs:line 54
   at Akka.TestKit.TestKitBase.<>c__DisplayClass119_0`1.<InternalExpectMsgEnvelope>b__2(T m, IActorRef sender) in C:\Users\brahm\source\repos\akka.net\src\core\Akka.TestKit\TestKitBase_Expect.cs:line 222
   at Akka.TestKit.TestKitBase.InternalExpectMsgEnvelope[T](Nullable`1 timeout, Action`2 assert, String hint, Boolean shouldLog) in C:\Users\brahm\source\repos\akka.net\src\core\Akka.TestKit\TestKitBase_Expect.cs:line 252
   at Akka.TestKit.TestKitBase.InternalExpectMsgEnvelope[T](Nullable`1 timeout, Action`1 msgAssert, Action`1 senderAssert, String hint) in C:\Users\brahm\source\repos\akka.net\src\core\Akka.TestKit\TestKitBase_Expect.cs:line 224
   at Akka.TestKit.TestKitBase.InternalExpectMsg[T](Nullable`1 timeout, Action`1 msgAssert, String hint) in C:\Users\brahm\source\repos\akka.net\src\core\Akka.TestKit\TestKitBase_Expect.cs:line 199
   at Akka.TestKit.TestKitBase.ExpectMsg[T](T message, Nullable`1 timeout, String hint) in C:\Users\brahm\source\repos\akka.net\src\core\Akka.TestKit\TestKitBase_Expect.cs:line 54
   at Akka.TestKit.Tests.Xunit2.TestEventListenerTests.AllTestForEventFilterBase`1.<ExpectAsync_should_await_actionAsync>b__22_1() in C:\Users\brahm\source\repos\akka.net\src\core\Akka.TestKit.Tests\TestEventListenerTests\AllTestForEventFilterBase.cs:line 183
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)

```

**Comment by brah-mcdude** [comment]
> Just need to `await actionAsync()` and delete the `Task.FromResult` call there I think.

I could not get rid of: "return Task.FromResult<object>(null); ". 
Maybe you can?

**Comment by Aaronontheweb** [comment]
resolved via #5538

**Comment by brah-mcdude** [comment]
Issue seems to appear in another scenario: 

This code: 

```c#
        protected async Task ExpectLogNoWarningsNorErrorsAsync(Func<Task> action)
        {
            await CreateEventFilter(Sys)
                .Custom(CustomErrorOrWarningFilter)
                .ExpectAsync(0, TimeSpan.FromSeconds(3),
                async () =>
                {
                    await action();
                });
        }
```

Invokes this code: 
```c#
        /// <summary>
        /// Async version of <see cref="Intercept{T}"/>
        /// </summary>
        protected Task<T> InterceptAsync<T>(Func<T> func, ActorSystem system, TimeSpan? timeout, int? expectedOccurrences, MatchedEventHandler matchedEventHandler = null)
        {
            return InterceptAsync(() => Task.FromResult(func()), system, timeout, expectedOccurrences, matchedEventHandler);
        }

```
Which seems to invoke: 
func()
without awaiting it.
